aspnetcore/performance/blazor/webassembly-best-practices.md

```diff
@@ -0,0 +1,156 @@
+
+## Use synchronous and unmarshalled JS interop APIs where appropriate
+
+Blazor WebAssembly offers two additional versions of <xref:Microsoft.JSInterop.IJSRuntime> over the single version available to Blazor Server apps:
+
+* <xref:Microsoft.JSInterop.IJSInProcessRuntime> allows invoking JS interop calls synchronously, which has less overhead than the asynchronous versions:
+
+  ```razor
+  @inject IJSRuntime JS
+
+  @code {
+      protected override void OnInitialized()
+      {
+          var jsInProcess = (IJSInProcessRuntime)JS;
+
+          var value = jsInProcess.Invoke<string>("jsInteropCall");
+      }
+  }
+  ```
+
+* <xref:Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime> permits unmarshalled JS interop calls:
+
+  ```javascript
+  function jsInteropCall() {
+    return BINDING.js_to_mono_obj("Hello world");
+  }
+  ```
+
+  ```razor
+  @inject IJSRuntime JS
+
+  @code {
+      protected override void OnInitialized()
+      {
+          var jsInProcess = (WebAssemblyJSRuntime)JS;
+
+          var value = jsInProcess.InvokeUnmarshalled<string>("jsInteropCall");
+      }
+  }
+  ```
+
+  > [!WARNING]
+  > While using <xref:Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime> has the least overhead of the JS interop approaches, the JavaScript APIs required to interact with these APIs are currently undocumented and subject to breaking changes in future releases.
+
+## Reduce app size
+
+### Intermediate Language (IL) linking
+
+[Linking a Blazor WebAssembly app](xref:host-and-deploy/blazor/configure-linker) reduces the app's size by trimming unused code in the app's binaries. By default, the linker is only enabled when building in `Release` configuration. Publish the app for deployment using the [dotnet publish](/dotnet/core/tools/dotnet-publish) command with the [-c|--configuration](/dotnet/core/tools/dotnet-publish#options) option set to `Release`:
+
+```dotnetcli
+dotnet publish -c Release
+```
+
+### Disable unused features
+
+Blazor WebAssembly's runtime includes the following .NET features that can be disabled if the app doesn't require them for a smaller payload size:
+
+* A data file is included to make timezone information correct. If the app doesn't require this feature, consider disabling it by setting the `BlazorEnableTimeZoneSupport` MSBuild property in the app's project file to `false`:
+
+  ```xml
+  <PropertyGroup>
+    <BlazorEnableTimeZoneSupport>false</BlazorEnableTimeZoneSupport>
+  </PropertyGroup>
+  ```
+
+* Collation information is included to make APIs such as [StringComparison.InvariantCultureIgnoreCase](xref:System.StringComparison.InvariantCultureIgnoreCase) work correctly. If you can verify that the app doesn't require the collation data, consider disabling it by setting the `BlazorWebAssemblyPreserveCollationData` MSBuild property in the app's project file to `false`:
+
+  ```xml
+  <PropertyGroup>
+    <BlazorWebAssemblyPreserveCollationData>
```

Comment:
I assume you added the line breaks to avoid horizontal scrolling in the doc. If possible, remove those line breaks. This sample looks odd.

Reply:
I did ... and I agree. 😁